### ISSUE TITLE
Multithreaded crashes Windows Explorer occationally

### ISSUE THREAD

**Issue by ToBos** [jingyu9575/multithreaded-download-manager#52]
The internal download manager of Firefox allows to change to the target folder. Sometimes the Windows Explorer flashes up for a very short time only and crashes. The Explorer window shows either a temporary file instead of the downloaded .rar archive or it seams empty. The second access to the target folder functions every time.

Firefox Quantum 65.0.2 (64Bit)
Multithreaded Download Manager 1.9.5.1
Windows 10 1809 (64Bit) Build 17763.316
TC4Shell 19.1.14

I don't know, whether this is a problem of Windows Explorer, the archiver TC4Shell, which integrates into Windows Explorer, or, whether it is a problem of the internal download manager of Firefox or of the add-on Multithreaded Download Manager.

How can I avoid this behaviour?

Regards Tobias

**Comment by jingyu9575** [comment]
Are you referring to this button "Open the containing folder" in the built-in download panel?

![default](https://user-images.githubusercontent.com/8923139/53857113-f627c780-400f-11e9-9b52-9baaf8e45af1.png)

**Comment by ToBos** [comment]
Yes. The first option of the right click menue (open containing folder) gives the same proble.
Windows Explorer crashes, not Firefox nor your add on
 
Von: jingyu9575 [mailto:notifications@github.com] 
Gesendet: Mittwoch, 6. März 2019 06:02
An: jingyu9575/multithreaded-download-manager <multithreaded-download-manager@noreply.github.com>
Cc: ToBos <tobias@bossert-inet.de>; Author <author@noreply.github.com>
Betreff: Re: [jingyu9575/multithreaded-download-manager] Multithreaded crashes Windows Explorer occationally (#52)
 
Are you referring to this button "Open the containing folder" in the built-in download panel?
 <https://user-images.githubusercontent.com/8923139/53857113-f627c780-400f-11e9-9b52-9baaf8e45af1.png>

**Comment by ToBos** [comment]
Is there any internal difference between the button [open containing folder] and the first option "open containing folder" in the context menue (right click)?
Up to now I allways used the right click menue. After your question I used the button 10 times without this problem. But the problem occured only sporadically, so nothing is proven at the moment. I guess the problem occured in about 10% of time.
 
Von: Tobias Bossert [mailto:tobias@bossert-inet.de] 
Gesendet: Mittwoch, 6. März 2019 11:11
An: 'jingyu9575/multithreaded-download-manager' <reply@reply.github.com>
Betreff: AW: [jingyu9575/multithreaded-download-manager] Multithreaded crashes Windows Explorer occationally (#52)
 
Yes. The first option of the right click menue (open containing folder) gives the same proble.
Windows Explorer crashes, not Firefox nor your add on
 
Von: jingyu9575 [mailto:notifications@github.com] 
Gesendet: Mittwoch, 6. März 2019 06:02
An: jingyu9575/multithreaded-download-manager <multithreaded-download-manager@noreply.github.com <mailto:multithreaded-download-manager@noreply.github.com> >
Cc: ToBos <tobias@bossert-inet.de <mailto:tobias@bossert-inet.de> >; Author <author@noreply.github.com <mailto:author@noreply.github.com> >
Betreff: Re: [jingyu9575/multithreaded-download-manager] Multithreaded crashes Windows Explorer occationally (#52)
 
Are you referring to this button "Open the containing folder" in the built-in download panel?
 <https://user-images.githubusercontent.com/8923139/53857113-f627c780-400f-11e9-9b52-9baaf8e45af1.png>

**Comment by jingyu9575** [comment]
> internal difference between the button [open containing folder] and the first option "open containing folder" in the context menu

I don't think there is any difference.

I'm not sure what is the reason. I tested in Firefox dev 66.0b13 but did not reproduce it.

Since Windows Explorer is crashing instead of Firefox, I think the problem may be within the explorer process i.e. TC4Shell. This extension can write large amount of data in a short time when the download has finished, and probably TC4Shell cannot deal with this.

**Comment by jingyu9575** [comment]
@ToBos (Please see GitHub for the edit. Also, GitHub is not recognizing your email's format and was showing the reply footer with your email address.)

**Comment by ToBos** [comment]
The problem persists, Windows Explorer crashes sometimes after clicking the button "open containing folder".
I used TC4Schell a long time before installing your add-on for Firefox. Windows Explorer never crashed before!
I can live with these crashes so far because no data are lost ...
 
Von: jingyu9575 [mailto:notifications@github.com] 
Gesendet: Donnerstag, 7. März 2019 12:26
An: jingyu9575/multithreaded-download-manager <multithreaded-download-manager@noreply.github.com>
Cc: ToBos <tobias@bossert-inet.de>; Author <author@noreply.github.com>
Betreff: Re: [jingyu9575/multithreaded-download-manager] Multithreaded crashes Windows Explorer occationally (#52)
 
internal difference between the button [open containing folder] and the first option "open containing folder" in the context menu
I don't think there is any difference.
I'm not sure what is the reason. I tested in Firefox dev 66.0b13 but did not reproduce it.
Since Windows Explorer is crashing instead of Firefox, I think the problem may be within the explorer process i.e. TC4Shell. This extension can write large amount of data in a short time when the download has finished, and probably TC4Shell cannot deal with this.
